feat: add features page on nsm site (#257)
networkservicemesh/site: content/docs/concepts/features/datapath_healing.md

<script type="text/javascript" src="https://viewer.diagrams.net/js/viewer-static.min.js"></script>

## Description

> The datapath healing is extremely simple. We use the existing connection to make sure our connection stays online. The connection is checked at specified intervals. As soon as the check fails, the client requests a new connection.

Reviewer: ```suggestion
Datapath healing is extremely simple. The existing connection is used to make
```

> The datapath healing is extremely simple. We use the existing connection to make sure our connection stays online. The connection is checked at specified intervals. As soon as the check fails, the client requests a new connection.

Reviewer: ```suggestion
Datapath healing is extremely simple. The existing connection is used to make sure the connection stays online. The connection is checked at specified intervals using 'ping'. As soon as the check fails, the client requests a new connection.
```

Author: Thanks, fixed!

Since NSM uses different data plane platforms the datapath liveness check could be differ for each platform.

## Parameters

These parameters are using for configuring datapath healing on the NSC side.

- `NSM_LIVENESS_CHECK_ENABLED` means datapath healing feature is enabled.
- `NSM_LIVENESS_CHECK_INTERVAL` means interval between pings.
- `NSM_LIVENESS_CHECK_TIMEOUT` means timeout for the ping.

## Benifits

- **Network Health:** Regular liveness checking can help monitor the health and availability of a network or a specific vWire. If the ping responses are consistently successful, it indicates that the network or device is functioning correctly. Any disruptions or failures in ping responses can alert the NSC to ask NSM to change the connection.

> - **Server Uptime:** For NSC, liveness checking a endoint's interface at regular intervals can be a way to ensure that the ednpoint is online and operational. If endpoint is not online then NSC requests a new endpoint *immediately*.

Reviewer 2: ```suggestion
- **Server Uptime:** For NSC, liveness checking a endpoint's interface at regular intervals can be a way to ensure that the endpoint is online and operational. If endpoint is not online then NSC requests a new endpoint *immediately*.
```

- **Intrusion Detection:** Regular Liveness Check can be part of a broader security monitoring strategy. It helps detect unauthorized changes or intrusions by alerting you to unexpected changes in liveness check response patterns.
- **Firewall and ACL Testing:** Liveness Check can be used to test the effectiveness of firewalls and Access Control Lists (ACLs). You can verify that the configured rules are working as intended if NSC doesn't come up.

## Add datapath healing for a custom dataplane framework

NSM uses the next contract for liveness check function.

```go
// LivenessCheck - function that returns true of conn is 'live' and false otherwise
type LivenessCheck func(deadlineCtx context.Context, conn *networkservice.Connection) bool
```

> To use a custom liveness check funciton you need simply implement and pass it to the heal client

Reviewer 2: ```suggestion
To use a custom liveness check function you need simply implement and pass it to the heal client
```
